build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+fi
+
+git commit -a -m "Glusterfs ${version} G.A (${flavor})"
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
+
+echo "Uploading the packages.."
+if [ "$os" == "ubuntu" ]; then
+        cd ..
+        dput ppa:gluster/glusterfs-${series} glusterfs_${version}-${os}1~${flavor}1_source.changes
+
+    echo "Done"
+    exit
+fi
+
+# we are using the same builder machine to build so we are running the "pbuilder
+# create" everytime to create the chroot according to the os and flavor we want to build.
+echo "creating chroot for ${os} ${flavor}"
+sudo pbuilder create --distribution ${flavor} --mirror ${mirror} --debootstrapopts --keyring=/usr/share/keyrings/${os}-archive-keyring.gpg
+
+echo "Building glusterfs-${version} for ${os} ${flavor} using the chroot and .dsc we created"
+
+# have to use the .dsc file inside the ${os}${flavor} folder
+sudo pbuilder build ~/${os}-${flavor}-Glusterfs-${version}/build/glusterfs_${version}-${release}.dsc | tee build.log
+
+#move the packages to packages directory.
+mv /var/cache/pbuilder/result/glusterfs*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
+if [ "$flavor" != "stretch" ]; then
+     mv /var/cache/pbuilder/result/libg*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
+fi
+/usr/share/debdelta/dpkg-sig -v -k ${pbuild_key} --sign builder ~/${os}-${flavor}-Glusterfs-${version}/packages/glusterfs-*${version}-${release}*.deb
+
+cd /var/www/repos/apt/debian/
+
+rm -rf pool/* dists/* db/*
+
+cp ~/conf.distributions/${series} conf/distributions
+
+# distribute Debian packages using apt
+for i in ~/${os}-${flavor}-Glusterfs-${version}/packages/glusterfs-*${version}-${release}*; do reprepro includedeb $flavor $i; done
+if [ "$flavor" != "stretch" ]; then
+     for i in ~/${os}-${flavor}-Glusterfs-${version}/packages/libg*${version}-${release}*.deb; do reprepro includedeb $flavor $i; done
+fi
+reprepro includedsc ${flavor} ~/${os}-${flavor}-Glusterfs-${version}/build/glusterfs_${version}-${release}.dsc
+
+tar czf ~/${os}-${flavor}-Glusterfs-${version}/${flavor}-apt-amd64-${version}.tgz pool/ dists/
+
+echo "Pushing Changelog changes.."
+git push origin ${flavor}-${series}-local:${flavor}-glusterfs-${series}
+
+cd ~/${os}-${flavor}-Glusterfs-${version}
+
+#copy the tar.gz file produced by the build to download.rht.gluster.org:/var/www/scratch
+scp $flavor-apt-amd64-$version.tgz gluster_ant@download.rht.gluster.org:/var/www/scratch
+
+ssh gluster_ant@download.rht.gluster.org
+
+cd /var/www/html/pub/gluster/glusterfs/$series/$version/$os/${flavor^}/amd64/apt
+sudo tar xpf /var/www/scratch/$flavor*$version.tar.gz
+
+#update latest inside the series you are building for
+ln -nf /var/www/html/pub/gluster/glusterfs/$series/$version/$os/$flavor/amd64/apt /var/www/html/pub/gluster/glusterfs/$series/LATEST
```

Comment:
If we capitalize the flavor with ${flavor^}, shouldn't we do the same here ?

Also, isn't a "-s" missing here ? that's for a directory, and hardlink are not autorized for them, if I am not wrong.

Reply:
fixed.